### DISCUSSION TITLE
Warning in MIfitHMM "working parameter standard errors are not finite for HMM fits"

### DISCUSSION THREAD

**Discussion by johnbergeron99** [bmcclintock/momentuHMM#179] [Q&A] [answered]
I have been working with GPS collar datasets in batches of roughly 4000 observations with 5 min sample intervals. My code reads as follows:

```
ln.prior <- function(theta) dnorm(theta[2],-4,2,log=TRUE)
crw <- crawlWrap(obsData = j,timeStep = "5 min",
                 Time.name = "time", coord = c("x","y"),
                 prior = ln.prior,
                 theta = c(5,3))
```

My crawl output is then piped into the `prepData` function:
```
bestData <- prepData(data=crw)
```

Which I use for finding best fit parameters via the following:
```
mu0 <- c(50,100)
sigma0 <- c(30,60)
stepPar0 <- c(mu0,sigma0)
dist <- list(step= "gamma", angle= "vm")
Par0 <- list(step= stepPar0, angle= c(1.65,4))`

bestFit <- fitHMM(data= bestData, nbStates= 2, 
                  dist= dist, 
                  Par0= Par0, 
                  prior = ln.prior)

bPar0 <- getPar(bestFit)
```
Using these parameters, I assigned `fixPar`. (Prior to doing so, the "Hessian is singular warning" appeared.)
```
fixPar <-  list(beta=bPar0$beta,
                delta=bPar0$delta)
```
Running the following `MIfitHMM` results in the warning from the title:
```
hmm <- MIfitHMM(miData = crw, nSims = 16, ncores = 16, nbStates= 2, 
                 dist= dist, 
                 Par0= bPar0$Par,
                 delta0 = bPar0$delta,
                 beta0 = bPar0$beta,
                 prior=ln.prior,
                 optMethod = "nlm",
                 nlmPar = list(hessian=T, print.level=2),
                 fixPar = fixPar)
```
The warning "NaNs produced: working parameter standard errors are not finite for HMM fits..." appears. I have found these NaN values to be the `hmm[["miSum"]][["Par"]][["beta"]][[angle]]`&`[[beta]]`&`[[delta]]` standard errors and confidence intervals. Also, the `hmm2[["miSum"]][["Par"]][["real"]][["angle"]]` standard errors and confidence intervals are `NA`, while the `hmm2[["miSum"]][["Par"]][["real"]][["gamma"]]` & `[[delta]]` standard errors are 0. 

Using `na.rm=T`, I can get finite pooled estimates from `MIfitHMM` for the successful 50-90% of simulations. Should I expect some proportion of my simulations to fail? If so, I am safe to proceed using `na.rm=T` and just  increasing the number of simulation to, let's say, 1000? If not, how might I go about resolving these warnings?

**Comment by bmcclintock** [comment, marked as answer]
Singular hessians often are indicative of issues with the model and/or data. See #175 and #177. Fixed parameters will always have zero standard errors (because they are fixed). I can't diagnose based on the information provided, but my best guess is that you have working scale parameters hitting against the boundary of the real parameter space (in which case you may want to fix them or use a prior to prevent them from wandering off into the nether regions of the working scale parameter space).
